### PR TITLE
Implement test system

### PR DESCRIPTION
**Pull request by coffeejunk**
Implement a test system to automatically verify the validity of the JSON Schemas and given test data.

This implements CredentialEngine/CredentialRegistry#96

### CODE REVIEW

**Comment by coffeejunk** [comment]
Since I don't have admin access to the repository yet, I could turn on Travis CI. However you can see the [result of this in my fork](https://travis-ci.org/coffeejunk/CTDL_Json_Validation_Schema/jobs/228279368)

File: spec/json_validation_spec.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    end
+  end
+
+  Dir[File.expand_path('../../test_data/**', __FILE__)].each do |resource_type|
+    context resource_type.slice(resource_type.index('test_data')..-1).to_s do
+      schema_name = resource_type.split('/').last.to_sym
+      resources   = Dir[resource_type + '/**']
+
+      context 'validates resources' do
+        resources.reject { |r| r =~ /invalid/ }.each do |resource|
```
Comment (leolara) [inline spec/json_validation_spec.rb]: @coffeejunk shouldnt directory be valid?

Comment (leolara) [inline spec/json_validation_spec.rb]: I mean  `|r| r =~ /valid/`

Comment (coffeejunk) [inline spec/json_validation_spec.rb:24]: No, I'm getting an array with all the resources, I then reject (i.e. clean up) everything that includes `invalid`, so the test resources don't necessarily have to have `valid` in their name.

in line :31 I do the opposite and reject everything that does not include `invalid`.

**Comment by leolara** [comment]
@coffeejunk Ok to merge

**Pull request merged**